plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -357,6 +358,49 @@ func TestGetRoundTripper(t *testing.T) {
 	}
 }
 
+func TestGetRoundTripperToken(t *testing.T) {
+	tokenFilePath := "test/test.txt"
+	wantBearer := "token from file"
+	logger := zap.NewNop()
+	// Create temp file with token if tokenFilePath is provided
+	dir, file := filepath.Split(tokenFilePath)
+	err := os.MkdirAll(dir, 0o750)
+	require.NoError(t, err)
+	err = os.WriteFile(filepath.Join(dir, file), []byte(wantBearer), 0o660)
```

Comment:
Instead of `filepath.Split(tokenFilePath)`, could we use `filepath.Dir(tokenFilePath)` to just get the `dir` so that we can recursively create the directory structure for the test file, then replace `filepath.Join(dir, file)` with just `tokenFilePath` (`file` isn't really needed)?